### PR TITLE
Cleanup: fix golint failures in pkg/kubelet/dockershim

### PR DESCRIPTION
**Pull request by ESWZY**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fix golint failures for package `pkg/kubelet/dockershim` and its sub pkg.

**Which issue(s) this PR fixes**:

Ref: #68026 [Fix golint failures](https://github.com/kubernetes/kubernetes/issues/68026)

Break up: #93222 [Cleanup: fix linting failures](https://github.com/kubernetes/kubernetes/pull/93222)

**Special notes for your reviewer**:

Change some type names and and pkg names to avoid golint failures. Also simplifyied some variable name to avoid: 

```
xxx.go:X:6: type name will be used as xxx.XxxID by other packages, and that stutters; consider calling this ID
```

If that caused ambiguity, please tell me to revert.

To break up another PR, I reverted some of commits. Sorry for a long commit log.

Thanks for your review! If you have any reply, I will respond as soon as possible!

### CODE REVIEW

**Comment by ESWZY** [comment]
/assign @vishh

Thx!

File: pkg/kubelet/dockershim/network/cni/cni.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func SplitDirs(dirs string) []string {
 }
 
 // ProbeNetworkPlugins : get the network plugin based on cni conf file and bin file
-func ProbeNetworkPlugins(confDir, cacheDir string, binDirs []string) []network.NetworkPlugin {
+func ProbeNetworkPlugins(confDir, cacheDir string, binDirs []string) []network.Plugin {
```
Comment (shubheksha) [inline pkg/kubelet/dockershim/network/cni/cni.go:133]: Fixing this makes the code less readable imo and it'll be good to leave this as is. What do you think?

File: pkg/kubelet/dockershim/network/hostport/hostport_manager.go [hunk 2]
```diff
@@ -38,8 +38,8 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-// HostPortManager is an interface for adding and removing hostport for a given pod sandbox.
-type HostPortManager interface {
+// Manager is an interface for adding and removing hostport for a given pod sandbox.
```
Comment (shubheksha) [inline pkg/kubelet/dockershim/network/hostport/hostport_manager.go:41]: Changing this makes the code less readable and I'd be tempted to leave it as is :)

File: pkg/kubelet/dockershim/network/hostport/hostport_syncer.go [hunk 3]
```diff
@@ -36,8 +36,8 @@ import (
 	utilnet "k8s.io/utils/net"
 )
 
-// HostportSyncer takes a list of PodPortMappings and implements hostport all at once
-type HostportSyncer interface {
+// Syncer takes a list of PodPortMappings and implements hostport all at once
```
Comment (shubheksha) [inline pkg/kubelet/dockershim/network/hostport/hostport_syncer.go:39]: Same comment as above here

File: pkg/kubelet/dockershim/network/plugins.go [hunk 4]
```diff
@@ -304,14 +304,14 @@ func (*NoopPortMappingGetter) GetPodPortMappings(containerID string) ([]*hostpor
 // proceed in parallel.
 type PluginManager struct {
 	// Network plugin being wrapped
-	plugin NetworkPlugin
+	plugin Plugin
```
Comment (shubheksha) [inline pkg/kubelet/dockershim/network/plugins.go:307]: Without `NetworkPlugin`, we dunno what kind of plugin it is

**Review by shubheksha — COMMENTED** [review]
Looks great overall - thanks for picking this up 🎉 
Go lint will complain about the stutter, but in some cases it's better for the variable name to stutter if it enhances code readability imo :)

**Comment by ESWZY** [comment]
> Looks great overall - thanks for picking this up 🎉
> Go lint will complain about the stutter, but in some cases it's better for the variable name to stutter if it enhances code readability imo :)

Thank you!

I changed those names just because the golint checking. If it's better to leave it as it is, I will revert that.

(Must need someme chanism to disable some golint chenk😂)

File: pkg/kubelet/dockershim/network/cni/cni.go (hunk 1 above)
Comment (ESWZY) [reply pkg/kubelet/dockershim/network/cni/cni.go:133]: Same as me

**Comment by SergeyKanzhelev** [comment]
dup of https://github.com/kubernetes/kubernetes/pull/91181?

**Comment by kikisdeliveryservice** [comment]
> dup of #91181?

adding a hold as there is a pre-existing PR that is already approved.

/hold

**Comment by SergeyKanzhelev** [comment]
#91181 was merged 
/close

**Comment by SergeyKanzhelev** [comment]
@ESWZY thank you for submitting this PR. Unfortunately it's not easy sometimes to find already opened PRs.

**Comment by ESWZY** [comment]
> @ESWZY thank you for submitting this PR. Unfortunately it's not easy sometimes to find already opened PRs.

Understand! Thank you! 😀

**Pull request closed without merging**